server/common/src/main/java/org/eclipse/lsp/cobol/common/mapping/TextMapReplacer.java

```diff
@@ -301,13 +235,37 @@ private void validateMapSize(String map) {
     }
   }
 
-  private void throwInvalidParameters(String header, String message, int line, int character) {
-    throw new IllegalArgumentException(
-        String.format("%s: %s. %s", header, message, generatePositionString(line, character)));
+  private void validateDocumentRange(ExtendedText extendedText, Range range, String entityName) {
+    List<ExtendedTextLine> lines = extendedText.getLines();
+    if (range.getStart().getLine() >= lines.size()) {
+      throw new IllegalArgumentException(entityName + "range start line out of bounds");
+    }
+    if (range.getStart().getCharacter() > lines.get(range.getStart().getLine()).size()) {
+      throw new IllegalArgumentException(entityName + "range start character out of bounds");
+    }
+    if (range.getEnd().getLine() >= lines.size()) {
+      throw new IllegalArgumentException(entityName + "range end line out of bounds");
+    }
```

Comment:
I think the equality might actually be valid if the last character in the text was newline.